src/NServiceBus.Core/Hosting/StartupDiagnostics/DiagnosticSettingsExtensions.cs

```diff
@@ -34,9 +49,12 @@ public static void CustomDiagnosticsWriter(this EndpointConfiguration config, Fu
             Guard.AgainstNull(nameof(config), config);
             Guard.AgainstNull(nameof(customDiagnosticsWriter), customDiagnosticsWriter);
 
-            config.Settings.Set<HostDiagnosticsWriter>(new HostDiagnosticsWriter(customDiagnosticsWriter));
+            config.Settings.Set("HostDiagnosticsWriter", customDiagnosticsWriter);
         }
 
-        internal const string DiagnosticsPathKey = "Diagnostics.RootPath";
+        internal static bool TryGetCustomDiagnosticsWriter(this ReadOnlySettings settings, out Func<string, Task> customDiagnosticsWriter)
```

Comment:
it feels a bit inconsistent that for some party of the extensions in this class we're using the internal key while for others we use a function instead.